From 040ae148b20231582524bbf5dae4e76f0640ee24 Mon Sep 17 00:00:00 2001
From: Martino Facchin <m.facchin@arduino.cc>
Date: Wed, 20 Oct 2021 11:03:09 +0200
Subject: [PATCH] rohm-bd718xx: add dt-configurable button keycode

---
 drivers/mfd/rohm-bd718x7.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/mfd/rohm-bd718x7.c b/drivers/mfd/rohm-bd718x7.c
index bdf8b38713cd..5dd9afdd5995 100644
--- a/drivers/mfd/rohm-bd718x7.c
+++ b/drivers/mfd/rohm-bd718x7.c
@@ -24,6 +24,7 @@ static struct gpio_keys_button button = {
 	.code = KEY_POWER,
 	.gpio = -1,
 	.type = EV_KEY,
+	.wakeup = 1,
 };
 
 static struct gpio_keys_platform_data bd718xx_powerkey_data = {
@@ -180,6 +181,13 @@ static int bd718xx_i2c_probe(struct i2c_client *i2c,
 
 	button.irq = ret;
 
+	unsigned int keycode;
+	ret = of_property_read_u32(bd718xx->chip.dev->of_node, "rohm,button-keycode",
+				   &keycode);
+	if (!ret) {
+		button.code = keycode;
+	}
+
 	ret = devm_mfd_add_devices(bd718xx->chip.dev, PLATFORM_DEVID_AUTO,
 				   bd718xx_mfd_cells,
 				   ARRAY_SIZE(bd718xx_mfd_cells), NULL, 0,
-- 
2.33.1

