From caaf98ac7eb7b5bdff4eb599cda0afcfb6209bf4 Mon Sep 17 00:00:00 2001
From: Massimo Pennazio <maxipenna@libero.it>
Date: Wed, 4 Aug 2021 17:06:21 +0200
Subject: [PATCH] Adding some useful debug prints

---
 adb/adb.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/adb/adb.c b/adb/adb.c
index 7d2eaec959..8c25304878 100644
--- a/adb/adb.c
+++ b/adb/adb.c
@@ -1370,8 +1370,10 @@ int adb_main(int is_daemon, int server_port)
     /* don't listen on a port (default 5037) if running in secure mode */
     /* don't run as root if we are running in secure mode */
     if (should_drop_privileges()) {
+        #if 0
         drop_capabilities_bounding_set_if_needed();

+        #endif
         /* then switch user and group to "shell" */
         if (setgid(1000) != 0) {
             exit(1);
@@ -1400,6 +1402,10 @@ int adb_main(int is_daemon, int server_port)
         // listen on USB
         usb_init();
         usb = 1;
+        fprintf(stderr, "Successfully opened %s or %s\n", USB_ADB_PATH, USB_FFS_ADB_EP0);
+    }
+    else {
+        fprintf(stderr, "Failed to access %s or %s\n", USB_ADB_PATH, USB_FFS_ADB_EP0);
     }
 
     // If one of these properties is set, also listen on that port
@@ -1436,6 +1442,8 @@ int adb_main(int is_daemon, int server_port)
     }
     D("Event loop starting\n");
 
+    fprintf(stderr, "Event loop starting\n");
+
     fdevent_loop();
 
     usb_cleanup();
-- 
2.25.1

