Index: kernel-source/drivers/pci/controller/dwc/pci-imx6.c
===================================================================
--- kernel-source.orig/drivers/pci/controller/dwc/pci-imx6.c
+++ kernel-source/drivers/pci/controller/dwc/pci-imx6.c
@@ -1170,6 +1170,7 @@ static void imx6_pcie_deassert_core_reset(struct imx6_pcie *imx6_pcie)
 	}
 
 	/* Some boards don't have PCIe reset GPIO. */
+#if 0
 	if (gpio_is_valid(imx6_pcie->reset_gpio)) {
 		gpio_set_value_cansleep(imx6_pcie->reset_gpio,
 					imx6_pcie->gpio_active_high);
@@ -1177,6 +1178,7 @@ static void imx6_pcie_deassert_core_reset(struct imx6_pcie *imx6_pcie)
 		gpio_set_value_cansleep(imx6_pcie->reset_gpio,
 					!imx6_pcie->gpio_active_high);
 	}
+#endif
 
 	switch (imx6_pcie->drvdata->variant) {
 	case IMX8QM:
@@ -1329,6 +1331,15 @@ static void imx6_pcie_deassert_core_reset(struct imx6_pcie *imx6_pcie)
 		break;
 	}
 
+#if 1
+	if (gpio_is_valid(imx6_pcie->reset_gpio)) {
+		gpio_set_value_cansleep(imx6_pcie->reset_gpio,
+					imx6_pcie->gpio_active_high);
+		msleep(100);
+		gpio_set_value_cansleep(imx6_pcie->reset_gpio,
+					!imx6_pcie->gpio_active_high);
+	}
+#endif
 	return;
 }
 
