From 0819bbbe39287d3aa0cabf927cb171f235da603d Mon Sep 17 00:00:00 2001
From: Jorge Ramirez-Ortiz <jorge@foundries.io>
Date: Mon, 1 Feb 2021 09:35:41 +0100
Subject: [PATCH] rtc: pcf8563: remove workaround

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>
---
 drivers/rtc/rtc-pcf8563.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/rtc/rtc-pcf8563.c b/drivers/rtc/rtc-pcf8563.c
index 17592fabfa40..9b1d65f7da9c 100644
--- a/drivers/rtc/rtc-pcf8563.c
+++ b/drivers/rtc/rtc-pcf8563.c
@@ -558,13 +558,6 @@ static int pcf8563_probe(struct i2c_client *client,
 	struct pcf8563 *pcf8563;
 	int err;
 	unsigned char buf;
-	static int defer = 2;
-
-	if (defer) {
-		defer--;
-		return -EPROBE_DEFER;
-	}
-
 
 	dev_dbg(&client->dev, "%s\n", __func__);
 
-- 
2.17.1

