Index: kernel-source/arch/arm64/boot/dts/arduino/anx7625.dtsi
===================================================================
--- kernel-source.orig/arch/arm64/boot/dts/arduino/anx7625.dtsi
+++ kernel-source/arch/arm64/boot/dts/arduino/anx7625.dtsi
@@ -20,7 +20,7 @@
 
 &i2c1 {
 
-	analogix_i2c@58{
+	analogix: analogix_i2c@58{
 		compatible ="analogix,anx7625";
 		status = "okay";
 		reg = <0x58>;
@@ -32,6 +32,7 @@
 		reset-gpios = <&gpio4 27 0x0>;      // VIDEO_RSTN
 		cbl_det-gpios = <&gpio4 22 0x0>;    // VIDEO_CABLE
 		intr_comm-gpios = <&gpio4 23 0x0>;  // VIDEO_ALT
+		#sound-dai-cells = <0>;
 
 		ports {
 			#address-cells = <1>;
@@ -55,9 +56,19 @@
 };
 
 / {
-	sound_card {
-		compatible = "audio-graph-card";
-		dais = <&sai2_port0>;
+	sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,name = "anx7625";
+		simple-audio-card,mclk-fs = <512>;
+		simple-audio-card,cpu {
+			sound-dai = <&sai2>;
+			dai-tdm-slot-num = <2>;
+			dai-tdm-slot-width = <32>;
+		};
+		simple-audio-card,codec {
+			sound-dai = <&analogix>;
+		};
 	};
 };
 
@@ -133,6 +144,7 @@
 	fsl,sai-multi-lane;
 	fsl,dataline,dsd = <0 0xff 0xff 2 0xff 0x11>;
 	dmas = <&sdma2 0 26 0>, <&sdma2 1 26 0>;
+	#sound-dai-cells = <0>;
 	status = "okay";
 
 	ports {
