From 00c7586ecbe18868a2a803666f2ef333d6fc7b6e Mon Sep 17 00:00:00 2001
From: Massimo Pennazio <maxipenna@libero.it>
Date: Mon, 18 Oct 2021 12:34:37 +0200
Subject: [PATCH 14/45] Adding some doc to pinmux at the end of dts file

---
 arch/arm/dts/portenta-m8.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/portenta-m8.dts b/arch/arm/dts/portenta-m8.dts
index 3258d0c72a..d8d4573255 100644
--- a/arch/arm/dts/portenta-m8.dts
+++ b/arch/arm/dts/portenta-m8.dts
@@ -436,9 +436,10 @@
 		>;
 	};
 
+	/* Some of these pins are leaved configured by internal ROM code */
 	pinctrl_hog_1: hoggrp-1 {
 		fsl,pins = <
-		MX8MM_IOMUXC_GPIO1_IO08_GPIO1_IO8			0x19
+			MX8MM_IOMUXC_GPIO1_IO08_GPIO1_IO8		0x19
 		>;
 	};
 };
-- 
2.35.1

