From ad89a7ef4711c1ee7b24a94471599b217b5c1f41 Mon Sep 17 00:00:00 2001
From: Massimo Pennazio <maxipenna@libero.it>
Date: Tue, 11 Jul 2023 10:46:22 +0000
Subject: [PATCH 24/24] Force Breakout carrier as default

Signed-off-by: Massimo Pennazio <maxipenna@libero.it>
---
 board/arduino/portenta-x8/portenta-x8.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/board/arduino/portenta-x8/portenta-x8.c b/board/arduino/portenta-x8/portenta-x8.c
index cde20616a06..c5c266d8081 100644
--- a/board/arduino/portenta-x8/portenta-x8.c
+++ b/board/arduino/portenta-x8/portenta-x8.c
@@ -519,6 +519,8 @@ static void setup_iomux_fec(void)
 	gpio_free(FEC_RST_PAD);
 }
 
+#define CONFIG_BREAKOUT_IS_DEFAULT
+
 int board_init(void)
 {
 	struct arm_smccc_res res;
-- 
2.34.1

