From 8a12b91c58e216d6234c9274bccf9f846c4337af Mon Sep 17 00:00:00 2001
From: Martino Facchin <m.facchin@arduino.cc>
Date: Thu, 24 Feb 2022 19:00:54 +0100
Subject: [PATCH 44/45] Forcefully restore FEC reset() with gpio

---
 board/arduino/portenta-x8/portenta-x8.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/board/arduino/portenta-x8/portenta-x8.c b/board/arduino/portenta-x8/portenta-x8.c
index cd74f75dab..d0bcb47597 100644
--- a/board/arduino/portenta-x8/portenta-x8.c
+++ b/board/arduino/portenta-x8/portenta-x8.c
@@ -443,6 +443,22 @@ int board_usb_cleanup(int index, enum usb_init_type init)
 #define DISPMIX				9
 #define MIPI				10
 
+#define FEC_RST_PAD       IMX_GPIO_NR(3, 6)
+
+static void setup_iomux_fec(void)
+{
+	printf("setup_iomux_fec\n");
+
+	/* reset PHY and ensure RX_DV/CLK125_EN is pulled high to enable TX_REF_CLK */
+	//imx_iomux_v3_setup_multiple_pads(fec1_rst_pads, ARRAY_SIZE(fec1_rst_pads));
+	gpio_request(FEC_RST_PAD, "fec1_rst");
+	gpio_direction_output(FEC_RST_PAD, 0);
+	udelay(500);
+	gpio_direction_output(FEC_RST_PAD, 1);
+	udelay(100000);
+	gpio_free(FEC_RST_PAD);
+}
+
 int board_init(void)
 {
 	struct arm_smccc_res res;
@@ -453,6 +469,8 @@ int board_init(void)
 	setup_typec();
 #endif
 
+	setup_iomux_fec();
+
 #if IS_ENABLED(CONFIG_FEC_MXC)
 	setup_fec();
 #endif
-- 
2.35.1

