From baf4b170b403f9611bb0d79259e54eb0e6717b6f Mon Sep 17 00:00:00 2001
From: Massimo Pennazio <maxipenna@libero.it>
Date: Thu, 28 Sep 2023 10:44:58 +0000
Subject: [PATCH] Defines moved to defconfig

---
 include/configs/portenta-x8.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/configs/portenta-x8.h b/include/configs/portenta-x8.h
index 329375c4fc7..f09aa79f4a7 100644
--- a/include/configs/portenta-x8.h
+++ b/include/configs/portenta-x8.h
@@ -210,9 +210,9 @@
 #endif
 
 /* Link Definitions */
-#define CONFIG_LOADADDR			0x40800000
+//#define CONFIG_LOADADDR			0x40800000
 
-#define CONFIG_SYS_LOAD_ADDR		CONFIG_LOADADDR
+//#define CONFIG_SYS_LOAD_ADDR		CONFIG_LOADADDR
 
 #define CONFIG_SYS_INIT_RAM_ADDR        0x40000000
 #define CONFIG_SYS_INIT_RAM_SIZE        0x200000
@@ -255,7 +255,7 @@
 #define CONFIG_SYS_FSL_USDHC_NUM	2
 #define CONFIG_SYS_FSL_ESDHC_ADDR       0
 
-#define CONFIG_SYS_MMC_IMG_LOAD_PART	1
+//#define CONFIG_SYS_MMC_IMG_LOAD_PART	1
 
 #ifdef CONFIG_FSL_FSPI
 #define FSL_FSPI_FLASH_SIZE		SZ_32M
-- 
2.25.1

